src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -406,48 +406,36 @@ def process_file_upload_batch_parameters(namespace):
     # 2. try to extract account name and container name from destination string
     from .storage_url_helpers import StorageResourceIdentifier
     identifier = StorageResourceIdentifier(namespace.destination)
-    if not identifier.is_url():
-        namespace.destination_share_name = namespace.destination
-    elif identifier.filename or identifier.directory:
-        raise ValueError('incorrect usage: destination must be a file share url')
-    else:
-        namespace.destination_share_name = identifier.share
+    if identifier.is_url():
+        if identifier.filename or identifier.directory:
```

Comment:
Actually, this *should* work if identifier is a directory. If you create a directory in your share called "uploaded" and you want everything to be uploaded there, you should be able to. Forcing the user to batch-upload only to a share is like saying you can only put files on your machine at the root directory... but on any drive you want!